src/test/java/io/supertokens/test/multitenant/StorageLayerTest.java

```diff
@@ -62,4 +75,1075 @@ public void normalConfigContinuesToWork() throws InterruptedException, IOExcepti
+
+        JsonObject userInfoHeaders = new JsonObject();
+        userInfoHeaders.add("user-header", new JsonPrimitive("user-header-val"));
+
+        ExecutorService es = Executors.newFixedThreadPool(1000);
+
+        AtomicBoolean pass = new AtomicBoolean(true);
+
+        for (int i = 0; i < 3000; i++) {
+            es.execute(() -> {
+                try {
+                    mtStorage.overwriteTenantConfig(new TenantConfig(
+                            new TenantIdentifier(null, null, null),
+                            new EmailPasswordConfig(true),
+                            new ThirdPartyConfig(true, new ThirdPartyConfig.Provider[]{
+                                    new ThirdPartyConfig.Provider(
+                                            "google",
+                                            "Google",
+                                            new ThirdPartyConfig.ProviderClient[]{
+                                                    new ThirdPartyConfig.ProviderClient(
+                                                            "web",
+                                                            "client-id",
+                                                            "client-secret",
+                                                            new String[]{"scope-1", "scope-2"},
+                                                            true,
+                                                            new JsonObject()
+                                                    )
+                                            },
+                                            "https://auth.example.com/auth",
+                                            authParams,
+                                            "https://auth.example.com/token",
+                                            tokenParams,
+                                            "https://auth.example.com/user",
+                                            userInfoParams,
+                                            userInfoHeaders,
+                                            "https://auth.example.com/jwks",
+                                            "https://auth.example.com",
+                                            true,
+                                            new ThirdPartyConfig.UserInfoMap(
+                                                    new ThirdPartyConfig.UserInfoMapKeyValue("id1", "email1", "email_verified1"),
+                                                    new ThirdPartyConfig.UserInfoMapKeyValue("id2", "email2", "email_verified2")
+                                            )
+                                    )
+                            }),
+                            new PasswordlessConfig(true),
+                            new JsonObject()
+                    ));
+                } catch (Exception e) {
+                    if (e.getMessage().toLowerCase().contains("the transaction might succeed if retried")) {
```

Comment:
```suggestion
```

Reply:
Have moved this test to postgres plugin.

Reply:
This test needs to be in the core only.